pkg/clients/v2/factory.go

```diff
@@ -272,6 +274,8 @@ func (c *CachingFactory) createCloudwatchClient(regionConfig *aws.Config) *cloud
 		if c.logger.IsDebugEnabled() {
 			options.ClientLogMode = aws.LogRequestWithBody | aws.LogResponseWithBody
 		}
+
+		// Setting an explicit retryer will override the default settings on the config
 		options.Retryer = retry.NewStandard(func(options *retry.StandardOptions) {
 			options.MaxAttempts = 5
 			options.MaxBackoff = 3 * time.Second
```

Comment:
I believe this is not impacted by the upstream bug since it's using `Retryer` instead of `RetryMaxAttempts` but it's really hard to test if that's true.